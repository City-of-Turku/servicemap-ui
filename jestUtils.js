import React from 'react';
import { render } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import { ThemeProvider } from '@mui/material/styles';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import themes from './src/themes';
import finnish from './src/i18n/fi';

const mockStore = configureStore([]);

// Mock props for intl provider
const intlMock = {
  locale: 'fi',
  messages: finnish,
  wrapRichTextChunksInFragment: false,
};

const Providers = (mockState) => ({ children }) => {
  const store = mockStore(mockState);
  return (
    <Provider store={store}>
      <IntlProvider {...intlMock}>
        <ThemeProvider theme={themes.SMTheme}>
          {children}
        </ThemeProvider>
      </IntlProvider>
    </Provider>
  );
}

export const getRenderWithProviders = mockState => component => render(component, { wrapper: Providers(mockState) });