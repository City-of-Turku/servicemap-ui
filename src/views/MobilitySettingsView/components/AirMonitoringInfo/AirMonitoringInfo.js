import { Link, Typography } from '@mui/material';
import PropTypes from 'prop-types';
import React, { useEffect, useState } from 'react';
import styled from '@emotion/styled';
import { useSelector } from 'react-redux';
import { useIntl } from 'react-intl';
import useLocaleText from '../../../../utils/useLocaleText';

const AirMonitoringInfo = ({ infoTexts }) => {
  const [url, setUrl] = useState(infoTexts.url.fi);
  const intl = useIntl();
  const locale = useSelector(state => state.user.locale);

  const getLocaleText = useLocaleText();

  useEffect(() => {
    setUrl(getLocaleText(infoTexts.url));
  }, [locale]);

  const text = textValue => (
    <StyledTypography variant="body2" component="p">
      {intl.formatMessage({
        id: textValue,
      })}
    </StyledTypography>
  );

  return (
    <StyledContainer>
      {text(infoTexts.paragraph1)}
      {text(infoTexts.paragraph2)}
      {text(infoTexts.paragraph3)}
      <Link target="_blank" href={url}>
        {text(infoTexts.link)}
      </Link>
    </StyledContainer>
  );
};

const StyledTypography = styled(Typography)(({ theme }) => ({
  marginBottom: theme.spacing(1),
}));

const StyledContainer = styled.div(({ theme }) => ({
  padding: theme.spacing(2),
  textAlign: 'left',
  paddingBottom: theme.spacing(1),
}));

AirMonitoringInfo.propTypes = {
  infoTexts: PropTypes.shape({
    paragraph1: PropTypes.string,
    paragraph2: PropTypes.string,
    paragraph3: PropTypes.string,
    link: PropTypes.string,
    url: PropTypes.shape({
      fi: PropTypes.string,
      en: PropTypes.string,
      sv: PropTypes.string,
    }),
  }),
};

AirMonitoringInfo.defaultProps = {
  infoTexts: {},
};

export default AirMonitoringInfo;
