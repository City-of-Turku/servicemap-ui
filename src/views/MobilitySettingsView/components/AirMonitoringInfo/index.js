import AirMonitoringInfo from './AirMonitoringInfo';

export default AirMonitoringInfo;
