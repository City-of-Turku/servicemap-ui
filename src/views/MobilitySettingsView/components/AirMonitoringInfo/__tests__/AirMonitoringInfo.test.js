// Link.react.test.js
import React from 'react';
import AirMonitoringInfo from '../index';
import { initialState } from '../../../../../redux/reducers/user';
import { getRenderWithProviders } from '../../../../../../jestUtils';
import finnishTranslations from '../../../../../i18n/fi';

const mockProps = {
  infoTexts: {
    paragraph1: 'mobilityPlatform.info.airMonitoring.paragraph.1',
    paragraph2: 'mobilityPlatform.info.airMonitoring.paragraph.2',
    paragraph3: 'mobilityPlatform.info.airMonitoring.paragraph.3',
    link: 'mobilityPlatform.info.airMonitoring.link',
    url: {
      fi: 'https://www.ilmatieteenlaitos.fi/ilmanlaatu',
      en: 'https://en.ilmatieteenlaitos.fi/air-quality',
      sv: 'https://sv.ilmatieteenlaitos.fi/luftkvalitet',
    },
  },
};

const renderWithProviders = getRenderWithProviders({
  user: initialState,
});

describe('<AirMonitoringInfo />', () => {
  it('should match snapshot', () => {
    const { container } = renderWithProviders(<AirMonitoringInfo {...mockProps} />);
    expect(container).toMatchSnapshot();
  });

  it('does show text correctly', () => {
    const { container } = renderWithProviders(<AirMonitoringInfo {...mockProps} />);

    const p = container.querySelectorAll('p');
    const link = container.querySelector('a');
    expect(p[0].textContent).toContain(finnishTranslations['mobilityPlatform.info.airMonitoring.paragraph.1']);
    expect(p[1].textContent).toContain(finnishTranslations['mobilityPlatform.info.airMonitoring.paragraph.2']);
    expect(p[2].textContent).toContain(finnishTranslations['mobilityPlatform.info.airMonitoring.paragraph.3']);
    expect(link.textContent).toContain(finnishTranslations['mobilityPlatform.info.airMonitoring.link']);
  });
});
