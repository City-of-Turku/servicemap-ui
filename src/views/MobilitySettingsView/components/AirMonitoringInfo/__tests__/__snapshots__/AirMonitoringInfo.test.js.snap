// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AirMonitoringInfo /> should match snapshot 1`] = `
<div>
  <div
    class="css-1ouci0q"
  >
    <p
      class="MuiTypography-root MuiTypography-body2 css-1qxgxsz-MuiTypography-root"
    >
      Ilmansaastetiedot näytetään kullekin mittausasemalle tunnin välein laskettavana indeksinä, ja se kattaa kaikki ks. mittausasemalla mitatut saasteet. Ilmanlaatu luokitellaan viiteen luokkaan: vihreä=hyvä, keltainen=tyydyttävä, oranssi=välttävä, punainen=huono, violetti=erittäin huono.
    </p>
    <p
      class="MuiTypography-root MuiTypography-body2 css-1qxgxsz-MuiTypography-root"
    >
      Laskennassa otetaan huomioon rikkidioksidi (SO2), typpidioksidi (NO2), hengitettävät hiukkaset (PM10), pienhiukkaset (PM2,5) sekä otsoni (O3). Kun ilmanlaatu on huono tai erittäin huono, terveysvaikutukset ovat mahdollisia herkillä ihmisillä. Ilmanlaatua heikentävät katupöly, liikenteen päästöt, puun pienpoltto, energiantuotanto sekä ajoittaiset kaukokulkeumat.
    </p>
    <p
      class="MuiTypography-root MuiTypography-body2 css-1qxgxsz-MuiTypography-root"
    >
      Palvelukartan ilmanlaatutiedot saadaan Turun seudun ilmansuojelun yhteistyöryhmältä.
    </p>
    <a
      class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-11g60ih-MuiTypography-root-MuiLink-root"
      href="https://www.ilmatieteenlaitos.fi/ilmanlaatu"
      target="_blank"
    >
      <p
        class="MuiTypography-root MuiTypography-body2 css-1qxgxsz-MuiTypography-root"
      >
        Lisätietoja saa osoitteesta https://www.ilmatieteenlaitos.fi/ilmanlaatu
      </p>
    </a>
  </div>
</div>
`;
